File: src/data-structures-and-algorithms-with-javascript-book/algorithms/bubbleSort.js
function bubbleSort(arr) {
    const len = arr.length
    let temp

    for(let i = 0; i < len - 1; ++i) {
        for(let j = 0; j < len - 1 - i; ++j) { // 注意这里i的含义
            if (arr[j] > arr[j + 1]) {
                temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp
            }
        }
    }

    return arr
}

const array = [61, 85, 19, 88, 68, 8, 70, 29]
const sortedArray = bubbleSort(array)

console.log(sortedArray)    // [ 8, 19, 29, 61, 68, 70, 85, 88 ]

File: src/03/bubblesort/bubblesort.go
package main

import (
    "fmt"
)

func bubbleSort(array []int) []int {
    length := len(array)

    for i := 0; i < length - 1; i++ {
        for j := 0; j < length - i - 1; j++ {
            if array[j] > array[j+1] {
                array[j], array[j+1] = array[j+1], array[j]
            }
        }
    }

    return array
}

func main() {
    arr := []int{12,32,52,1,3,56,74,24,8}
    fmt.Println(bubbleSort(arr))
}
